games-puzzle/kpat/kpat-17.12.0.recipe

```diff
@@ -0,0 +1,140 @@
+PATCHES="kpat-$portVersion.patchset"
+ADDITIONAL_FILES="kpat.rdef.in"
+	
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	kpat$secondaryArchSuffix = $portVersion
+	app:kpat$secondaryArchSuffix
+	lib:libkcardgame$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	cmd:dbus_launch
+	lib:libKF5Attica$secondaryArchSuffix
+	lib:libKF5Auth$secondaryArchSuffix
+	lib:libKF5Bookmarks$secondaryArchSuffix
+	lib:libKF5Codecs$secondaryArchSuffix
+	lib:libKF5Completion$secondaryArchSuffix
+	lib:libKF5ConfigCore$secondaryArchSuffix
+	lib:libKF5ConfigWidgets$secondaryArchSuffix
+	lib:libKF5CoreAddons$secondaryArchSuffix
+	lib:libKF5Crash$secondaryArchSuffix
+	lib:libKF5DBusAddons$secondaryArchSuffix
+	lib:libKF5GuiAddons$secondaryArchSuffix
+	lib:libKF5I18n$secondaryArchSuffix
+	lib:libKF5ItemViews$secondaryArchSuffix
+	lib:libKF5JobWidgets$secondaryArchSuffix
+	lib:libKF5KDEGames$secondaryArchSuffix
+	lib:libKF5KIOCore$secondaryArchSuffix
+	lib:libKF5NewStuff$secondaryArchSuffix
+	lib:libKF5Service$secondaryArchSuffix
+	lib:libKF5WidgetsAddons$secondaryArchSuffix
+	lib:libKF5XmlGui$secondaryArchSuffix
+	lib:libQt5Core$secondaryArchSuffix
+	lib:libQt5Gui$secondaryArchSuffix
+	lib:libQt5Qml$secondaryArchSuffix
+	lib:libQt5Quick$secondaryArchSuffix
+	lib:libQt5Widgets$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	extra_cmake_modules >= 5.39
+	haiku${secondaryArchSuffix}_devel
+	devel:libKF5Attica$secondaryArchSuffix
+	devel:libKF5Auth$secondaryArchSuffix
+	devel:libKF5Bookmarks$secondaryArchSuffix
+	devel:libKF5Codecs$secondaryArchSuffix
+	devel:libKF5Completion$secondaryArchSuffix
+	devel:libKF5ConfigCore$secondaryArchSuffix
+	devel:libKF5ConfigWidgets$secondaryArchSuffix
+	devel:libKF5CoreAddons$secondaryArchSuffix
+	devel:libKF5Crash$secondaryArchSuffix
+	devel:libKF5DBusAddons$secondaryArchSuffix
+	devel:libKF5GuiAddons$secondaryArchSuffix
+	devel:libKF5I18n$secondaryArchSuffix
+	devel:libKF5ItemViews$secondaryArchSuffix
+	devel:libKF5JobWidgets$secondaryArchSuffix
+	devel:libKF5KDEGames$secondaryArchSuffix
+	devel:libKF5KIOCore$secondaryArchSuffix
+	devel:libKF5NewStuff$secondaryArchSuffix
+	devel:libKF5Service$secondaryArchSuffix
+	devel:libKF5Solid$secondaryArchSuffix
+	devel:libKF5WidgetsAddons$secondaryArchSuffix
+	devel:libKF5XmlGui$secondaryArchSuffix
+	devel:libQt5Core$secondaryArchSuffix
+	devel:libQt5Gui$secondaryArchSuffix
+	devel:libQt5Qml$secondaryArchSuffix
+	devel:libQt5Quick$secondaryArchSuffix
+	devel:libQt5Widgets$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:cmake
+	cmd:g++$secondaryArchSuffix
+	cmd:gettext$secondaryArchSuffix
+	cmd:make
+	cmd:python2
+	cmd:update_mime_database$secondaryArchSuffix
+	"
+
+BUILD()
+{
+	mkdir -p build
+	cd build
+
+	cmake .. \
+		-DECM_DIR=/system/data/cmake/Modules/ECM/cmake \
+		-DCMAKE_INSTALL_PREFIX=$appsDir/kpat \
+		-DKDE_INSTALL_BINDIR=$appsDir/kpat \
+		-DKDE_INSTALL_LIBDIR=$appsDir/kpat/lib \
```

Comment:
Please use `$libDir` instead

```diff
-		-DKDE_INSTALL_LIBDIR=$appsDir/kpat/lib \
+		-DKDE_INSTALL_LIBDIR=$libDir \

Reply:
If this library isn't used by other apps I think it's better to live in kpat folder.

Reply:
The app can't find it anyway if it's installed in "$libDir", it doesn't work.

Reply:
I did not test it, but I wonder if it would work with these changes:
```diff
- 		-DCMAKE_INSTALL_PREFIX=$appsDir/kpat \
+ 		-DCMAKE_INSTALL_PREFIX=$prefix \
-		-DKDE_INSTALL_BINDIR=$appsDir/kpat \
+		-DCMAKE_INSTALL_BINDIR=$appsDir \
-		-DKDE_INSTALL_LIBDIR=$appsDir/kpat/lib \
+		-DCMAKE_INSTALL_LIBDIR=$libDir \
```
(You would also need to modify `INSTALL()`.)

Reply:
It doesn't really matter whether you use -DCMAKE or -DKDE i tried both, either way the app seems only to look for it in $appsDir/KPatience/lib.